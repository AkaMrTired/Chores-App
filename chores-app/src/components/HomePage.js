import React from "react";
import Login from "./Login";

const HomePage = () => {
  return (
    <div className="container">
      <h1>Home Page</h1>
      <Login />
      <p>Not yet a member?</p>
      <button type="button">
        <a href="/signup">Sign Up!</a>
      </button>
    </div>
  );
};
export default HomePage;
