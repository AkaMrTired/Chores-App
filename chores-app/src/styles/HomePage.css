h1 {
  text-align: center;
  color: var(--purple-dark);
}

.span-bish {
  font-weight: bold;
  color: var(--peach);
}

.span-bash {
  font-weight: bold;
  color: var(--pink);
}
.span-dosh {
  font-weight: bold;
  color: var(--purple);
}

h3 {
  text-align: center;
  color: var(--gray);
  margin-bottom: 2rem;
}

p {
  text-align: center;
  margin-top: 3rem;
}
.home-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
