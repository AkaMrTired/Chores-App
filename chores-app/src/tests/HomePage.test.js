import { render, screen } from "@testing-library/react";
import React from "react";
import HomePage from "../components/HomePage";

describe("the homepage", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<HomePage />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<HomePage />);
    expect(screen.getByText(/homepage/i)).toBeInTheDocument();
    expect(screen.getByText(/login/i)).toBeInTheDocument();
    expect(screen.getByText(/name/i)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /Log In/i })).toBeInTheDocument();
    expect(screen.getByText(/password/i)).toBeInTheDocument();
    expect(screen.getByText(/not yet a member/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /Sign Up/i })
    ).toBeInTheDocument();
  });
});
