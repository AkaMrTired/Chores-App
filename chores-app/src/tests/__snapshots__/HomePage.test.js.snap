// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the homepage matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Home Page
    </h1>
    <div
      class="container"
    >
      <h1>
        Login
      </h1>
      <form
        id="login"
      >
        <label>
          Name:
          <input
            name="name"
            type="text"
          />
        </label>
        <label>
          Password:
          <input
            password="password"
            type="text"
          />
        </label>
        <input
          type="submit"
          value="Log in"
        />
      </form>
    </div>
    <p>
      Not yet a member?
    </p>
    <button
      type="button"
    >
      <a
        href="/signup"
      >
        Sign Up!
      </a>
    </button>
  </div>
</DocumentFragment>
`;
